src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -288,6 +289,37 @@ class StrawmanTest {
     println(xs17.to(List))
   }
 
+  def sortedSets(xs: immutable.SortedSet[Int]): Unit = {
+    val xs1 = xs.map((x: Int) => x.toString)
+    val xs2: immutable.SortedSet[String] = xs1
+  }
+
+  def mapOps(xs: Map[Int, String]): Unit = {
+    val xs1 = xs.map((k, v) => (v, k))
+    val xs2: strawman.collection.Map[String, Int] = xs1
+    val xs3 = xs.map(kv => (kv._2, kv._1))
+    val xs4: strawman.collection.Iterable[(String, Int)] = xs3
+    println(xs1)
+    println(xs2)
+    println(xs3)
+    println(xs4)
+  }
+
+  def sortedMaps(xs: immutable.SortedMap[String, Int]): Unit = {
+    val x1 = xs.get("foo")
+    val x2: Option[Int] = x1
+    val xs1 = xs + ("foo", 1)
+    val xs2: immutable.SortedMap[String, Int] = xs1
+    val xs3 = xs.map(kv => kv._1)
+    val xs4: immutable.Iterable[String] = xs3
+    val xs5 = xs.map((k: String, v: Int) => (v, k))
+    val xs6: immutable.SortedMap[Int, String] = xs5
+    class Foo
+//    val xs7 = xs.map((k: String, v: Int) => (new Foo, v)) Error: No implicit Ordering defined for Foo
+    val xs7 = (xs: immutable.Map[String, Int]).map((k, v) => (new Foo, v))
+    val xs8: immutable.Map[Foo, Int] = xs7
```

Comment:
This pattern works when classes don't have type parameters of their own:
```scala
trait Foo; trait Bar; trait TypeClass[C] {}
class Super {
  def foo[A](f: () => A): Foo = ???
}
class Sub extends Super {
  def foo[A: TypeClass](f: () => A): Bar = ???
}
```
If that pattern would work for collections it would be adequate to handle things like "this has to be a Tuple2 to stay a Map".  Right now you get "missing parameter type" on `(new Sub[String]).foo(_ + ".")` if you give the classes type parameters.  There's no real reason that has to be the case, though, AFAICT.